modules/upgrade/partials/incompat-changes.adoc

```diff
@@ -0,0 +1,9 @@
+=== Review incompatible changes
+
+* Starting in version 24.2, when managing configuration properties using the AlterConfigs API directly, or the `kafka-configs.sh` shell script without incremental flags, Redpanda resets all unspecified values to the default values. This aligns more closely with the behavior in Apache Kafka. There is no change if you're managing your configuration with tools like `rpk`, Redpanda Console, Kubernetes, Helm, or Terraform. 
```

Comment:
> or the `kafka-configs.sh` shell script without incremental flags,

This part is incorrect as far as I am aware. I've been testing with`kafka-configs.sh` locally and it has been using the IncrementalAlterConfigs API and not the AlterConfigs API since at least version 2.5.0 and there is no incremental flag that I am aware of. Pre-2.5.0 versions require a zookeeper endpoint for topic config updates, so they can't be used with redpanda. So whoever's managing redpanda topic configs with `kafka-configs.sh` is using the IncrementalAlterConfigs API.

I am curious where the quoted part came from, but I think we should remove it from this section.

Reply:
Thanks @pgellert! I can get this updated. 
cc @mattschumpert

Reply:
Ahh OK. @pgellert  I swear @twmb and I saw a version of this .sh with some --inc or --incremental flags and @twmb was the one saying at first that kafka-config.sh uses AlterConfigs.  SO would request @twmb to clarify here

Reply:
My knowledge for this comes from https://cwiki.apache.org/confluence/display/KAFKA/KIP-1011%3A+Use+incrementalAlterConfigs+when+updating+broker+configs+by+kafka-configs.sh

However, I didn't read closely enough -- it looks like modifying only _broker_ configs uses the old AlterConfigs, and modifying anything else does in fact use IncrementalAlterConfigs. My mistake.